services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -523,22 +524,18 @@ async def delete_file(
                 if not can.delete:
                     raise FileAccessRightError(access_right="delete", file_id=file_id)
 
-        with suppress(FileMetaDataNotFoundError):
-            # NOTE: deleting might be slow, so better ensure we release the connection
-            async with self.engine.acquire() as conn:
-                file: FileMetaDataAtDB = await db_file_meta_data.get(
-                    conn, TypeAdapter(SimcoreS3FileID).validate_python(file_id)
-                )
-            await get_s3_client(self.app).delete_objects_recursively(
-                bucket=file.bucket_name,
-                prefix=(
-                    ensure_ends_with(file.file_id, "/")
-                    if file.is_directory
-                    else file.file_id
-                ),
-            )
-            async with self.engine.acquire() as conn:
-                await db_file_meta_data.delete(conn, [file.file_id])
+        await get_s3_client(self.app).delete_objects_recursively(
+            bucket=self.simcore_bucket_name,
+            prefix=file_id,
+        )
+
+        async with self.engine.acquire() as conn:
+            if await try_get_fmd(conn, file_id):
+                await db_file_meta_data.delete(conn, [file_id])
+            if parent_dir_file_id := await get_directory_file_id(conn, file_id):
+                parent_dir_fmd = await db_file_meta_data.get(conn, parent_dir_file_id)
+                parent_dir_fmd.file_size = UNDEFINED_SIZE
+                await db_file_meta_data.upsert(conn, parent_dir_fmd)
```

Comment:
Instead of reading and writing the entire table entry I would only alter the file_size column.

Reply:
it does an upsert. so it actually updates what is necessary only